lib/Model/filter.js

```diff
@@ -23,18 +23,29 @@ Model.INITS.push(function(model) {
 
 function parseFilterArguments(model, args) {
   var fn = args.pop();
-  var options;
-  if (!model.isPath(args[args.length - 1])) {
+  var options, inputPaths;
+  var path = model.path(args.shift());
+
+  var last = args[args.length - 1];
+  if (!model.isPath(last) && !Array.isArray(last)) {
     options = args.pop();
   }
-  var path = model.path(args.shift());
-  var i = args.length;
+  if (args.length === 1 && Array.isArray(args[0])) {
+    // inputPaths provided as one array:
+    //   model.filter(inputPath, [inputPath1, inputPath2], fn)
+    inputPaths = args[0];
+  } else {
+    // inputPaths provided as var-args:
+    //   model.filter(inputPath, inputPath1, inputPath2, fn)
+    inputPaths = args;
+  }
+  var i = inputPaths.length;
   while (i--) {
-    args[i] = model.path(args[i]);
+    inputPaths[i] = model.path(inputPaths[i]);
   }
   return {
     path: path,
-    inputPaths: (args.length) ? args : null,
+    inputPaths: inputPaths,
```

Comment:
Previously, if there were no additional input paths, this would return `null`. I'm not certain if it matters, but let's play it safe and keep it that way.